opt/scopeserver/dataserver/modules/gserver/GServer.py

```diff
@@ -607,35 +608,20 @@ def getMarkerGenes(self, request, context):
         if "clusterMarkerMetrics" in md_clustering.keys():
             md_cmm = md_clustering["clusterMarkerMetrics"]
 
-            def create_cluster_marker_metric(metric):
-                return loom.get_cluster_marker_metrics(
-                    clustering_id=request.clusteringID, cluster_id=request.clusterID, metric_accessor=metric["accessor"]
-                )
-
             def protoize_cluster_marker_metric(metric):
                 return s_pb2.MarkerGenesMetric(
                     accessor=metric["accessor"],
                     name=metric["name"],
                     description=metric["description"],
-                    values=cluster_marker_metrics_final[metric["accessor"]],
-                )
-
-            def merge_cluster_marker_metrics(metrics):
-                return functools.reduce(
-                    lambda left, right: pd.merge(left, right, left_index=True, right_index=True, how="outer"), metrics,
+                    values=cluster_marker_metrics[metric["accessor"]],
                 )
 
-            cluster_marker_metrics = merge_cluster_marker_metrics(
-                metrics=[create_cluster_marker_metric(x) for x in md_cmm]
+            cluster_marker_metrics = loom.get_cluster_marker_table(
+                clusteringID=request.clusteringID, clusterID=request.clusterID
```

Comment:
This call eventually leads to calling `get_meta_data_clustering_by_id` which requires the config's dataHashSecret to validate Annotations. Not passing it down the chain means that the backend bails out when getting the marker table.
```suggestion
            cluster_marker_metrics = loom.get_cluster_marker_table(
                clusteringID=request.clusteringID, clusterID=request.clusterID, secret=self.config['dataHashSecret']
```